### ISSUE TITLE
Instance-based interface generates bogus connections in Power Grids

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#11310]
### Steps to Reproduce

- Download the development version of PowerGrids from https://github.com/PowerGrids/PowerGrids/tree/develop
- Open PowerGrids.Examples.IEEE14bus.ControlledGeneratorIEEE
- Delete the connection between the GEN component and the terminalAC top-level connector
- Connect the GEN component and the terminalAC connector again

OMEdit generates the following connect equation:
```modelica
  connect(GEN.terminalAC, terminalAC.terminalAC) annotation(Line(points = {{-26, -10}, {-26, -50}}));
```
which is bogus because there is no `terminalAC.terminalAC` connector to connect to. Then, of course, it complains that it is indeed bogus:
```
[3] 10:29:22 Scripting Error
Unable to find component terminalAC.terminalAC while parsing connect(GEN.terminalAC, terminalAC.terminalAC)
in PowerGrids.Examples.IEEE14bus.ControlledGeneratorIEEE.
```
What is also really bad is that the connection is not drawn, but the bogus connect statement remains in the model code, and if one tries many times to connect, the bogus connect statements pile up neatly.

### Expected Behavior

When connecting GEN to terminalAC, the following line should be generated:
```modelica
  connect(GEN.terminalAC, terminalAC) annotation(Line(points = {{-26, -10}, {-26, -50}}));
```

**Comment by perost** [comment]
It's not an issue with the instance API, the same thing happens with the old API. OMEdit simply calls:
```
addConnection(GEN.terminalAC,terminalAC.terminalAC,PowerGrids.Examples.IEEE14bus.ControlledGeneratorIEEE,annotate=Line(points={{-26,-10},{-26,-50}}))
```

**Comment by perost** [comment]
I should maybe also mention that you get different behaviour with the old and new API regarding error messages and what's visible, but the same connect equation is created regardless which is the real issue.

**Comment by casella** [comment]
OK, I'll leave it to @adeas31 then.

**Comment by adeas31** [comment]
#11316 fixes this. But now when I connect GEN component and the terminalAC connector it says,

`Connectors GEN.terminalAC and terminalAC are not compatible.`

@perost says that,

> They're the exact same type, but the connector contains inputs which is what the issue is I guess.
> Hmm, it might be that we don't check if the connectors are inside or outside. It's fine to connect input to input if one is inside and one is outside.

I am not sure if this is expected or should we fix it. In the later case make a new issue.

**Comment by perost** [comment]
> #11316 fixes this. But now when I connect GEN component and the terminalAC connector it says,
> 
> `Connectors GEN.terminalAC and terminalAC are not compatible.`
> 
> @perost says that,
> 
> > They're the exact same type, but the connector contains inputs which is what the issue is I guess.
> > Hmm, it might be that we don't check if the connectors are inside or outside. It's fine to connect input to input if one is inside and one is outside.
> 
> I am not sure if this is expected or should we fix it. In the later case make a new issue.

Fixed in #11317. The issue was that we only considered inside/outside when checking input/output of the connectors, but not when checking the components inside the connectors.

**Comment by adeas31** [comment]
Ported to maintenance 1.22
